macros/src/codegen.rs

```diff
@@ -485,7 +499,7 @@ pub fn generate_code(sm: &ParsedStateMachine) -> proc_macro2::TokenStream {
 
         /// State machine structure definition.
         pub struct StateMachine<#state_lifetimes_code T: StateMachineContext> {
-            state: States <#state_lifetimes_code>,
+            state: Option<States <#state_lifetimes_code>>,
```

Comment:
Why is the state being changed to an option? It looks like these changes may have slipped in to the reference changes?

Reply:
I need a way to move out the owned value from the state, that is why `Option` was added.